cmd/node/multi_node.go

```diff
@@ -0,0 +1,260 @@
+func NewRPCPoetHarnessClient() (*activation.RPCPoetClient, error) {
+	cfg, err := integration.DefaultConfig()
+	if err != nil {
+		return nil, err
+	}
+
+	cfg.N = 10
+	cfg.InitialRoundDuration = time.Duration(3 * time.Second).String()
+
+	h, err := integration.NewHarness(cfg)
+	if err != nil {
+		return nil, err
+	}
+
+	return activation.NewRPCPoetClient(h.PoetClient, h.TearDown), nil
+}
+
+// GracefulShutdown stops the current services running in apps
+func GracefulShutdown(apps []*SpacemeshApp) {
+	log.Info("Graceful shutdown begin")
+
+	var wg sync.WaitGroup
+	for _, app := range apps {
+		wg.Add(1)
+		go func(app *SpacemeshApp) {
+			app.stopServices()
+			wg.Done()
+		}(app)
+	}
+	wg.Wait()
+
+	log.Info("Graceful shutdown end")
+}
+
+//initialize a network mock object to simulate network between nodes.
+var net = service.NewSimulator()
+
+// InitSingleInstance initializes a node instance with given
+// configuration and parameters, it does not stop the instance.
+func InitSingleInstance(cfg config.Config, i int, genesisTime string, rng *amcl.RAND, storePath string, rolacle *eligibility.FixedRolacle, poetClient *activation.RPCPoetClient, fastHare bool, clock TickProvider) (*SpacemeshApp, error) {
+
+	smApp := NewSpacemeshApp()
+	smApp.Config = &cfg
+	smApp.Config.CoinbaseAccount = strconv.Itoa(i + 1)
+	smApp.Config.GenesisTime = genesisTime
+	edSgn := signing.NewEdSigner()
+	pub := edSgn.PublicKey()
+
+	vrfPriv, vrfPub := BLS381.GenKeyPair(rng)
+	vrfSigner := BLS381.NewBlsSigner(vrfPriv)
+	nodeID := types.NodeId{Key: pub.String(), VRFPublicKey: vrfPub}
+
+	swarm := net.NewNode()
+	dbStorepath := storePath
+
+	hareOracle := oracle.NewLocalOracle(rolacle, 5, nodeID)
+	hareOracle.Register(true, pub.String())
+
+	postClient, err := activation.NewPostClient(&smApp.Config.POST, util.Hex2Bytes(nodeID.Key))
+	if err != nil {
+		return nil, err
+	}
+
+	err = smApp.initServices(nodeID, swarm, dbStorepath, edSgn, false, hareOracle, uint32(smApp.Config.LayerAvgSize), postClient, poetClient, vrfSigner, uint16(smApp.Config.LayersPerEpoch), clock)
+	if err != nil {
+		return nil, err
+	}
+	smApp.setupGenesis()
+
+	return smApp, err
+}
+
+// Starts the run of a number of nodes, running in process consensus between them.
+// this also runs a single transaction between the nodes.
+func StartMultiNode(numOfinstances, layerAvgSize int, runUntilLayer uint32, dbPath string) {
+
+	runTillLayer := runUntilLayer
+
+	cfg := getTestDefaultConfig()
+	cfg.LayerAvgSize = layerAvgSize
+	numOfInstances := numOfinstances
+
+	dst := types.BytesToAddress([]byte{0x02})
+	acc1Signer, err := signing.NewEdSignerFromBuffer(util.FromHex(apiCfg.Account1Private))
+	if err != nil {
+		log.Panic("Could not build ed signer err=%v", err)
+	}
+	tx, err := mesh.NewSignedTx(0, dst, 10, 1, 1, acc1Signer)
+	if err != nil {
+		log.Panic("panicked creating signed tx err=%v", err)
+	}
+	txbytes, _ := types.InterfaceToBytes(tx)
+	path := dbPath + time.Now().Format("2006-01-02T15:04:05-0700")
```

Comment:
you can replace format date with `time.RFC3339`

Reply:
dont you think this is clearer?

Reply:
not really. this format has a name. if you look it up (or read the const value) you'll see the format.